### PR TITLE
Poap package

### PR DESCRIPTION
**Pull request by rlajous**
## Description

Please provide a short description of the changes in this pull request.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/poap-xyz/poap.js/blob/main/documentation/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/poaps/backend/src/methods/fetch_multiple_poaps.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export const fetch_multiple_poaps = async (
+  client: PoapsClient,
+): Promise<void> => {
+  try {
+    const data: PaginatedResult<POAP> = await client.fetch({
+      order: 'asc',
+      limit: 10,
+      offset: 1,
+    });
+    console.log(data);
```
Comment (jm42) [inline examples/poaps/backend/src/methods/fetch_multiple_poaps.ts:13]: What does this return? Maybe adding a comment saying what is to be expected from the response. Something like:

> The first 10 POAP tokens minted.

File: examples/poaps/backend/src/methods/fetch_single_poap.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export const fetch_single_poap = async (client: PoapsClient): Promise<void> => {
+  try {
+    const data: PaginatedResult<POAP> = await client.fetch({
+      order: 'asc',
+      limit: 10,
+      offset: 0,
+      ids: [1],
+    });
+    console.log(data);
```
Comment (jm42) [inline examples/poaps/backend/src/methods/fetch_single_poap.ts:12]: Same here, would be nice to add some description and let know that even if is paginated, there will be zero or one POAP instances on the response.

File: packages/poaps/src/types/input.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export interface FetchPoapsInput {
+  limit: number;
+  offset: number;
+  order?: string;
+  name?: string;
+  chain?: string;
+  idOrder?: string;
```
Comment (jm42) [inline packages/poaps/src/types/input.ts:7]: What's this for? Why has different case than the other fields?

File: packages/poaps/src/types/input.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  from?: string;
+  to?: string;
```
Comment (jm42) [inline packages/poaps/src/types/input.ts:9]: What does from/to mean? Are these tokens or transfers?

File: packages/poaps/src/types/input.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export interface FetchPoapsInput {
+  limit: number;
+  offset: number;
+  order?: string;
+  name?: string;
+  chain?: string;
```
Comment (jm42) [inline packages/poaps/src/types/input.ts:6]: Would be nicer if this was an enum.

File: packages/poaps/src/types/input.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+export interface FetchPoapsInput {
+  limit: number;
+  offset: number;
+  order?: string;
```
Comment (jm42) [inline packages/poaps/src/types/input.ts:4]: What can be the values here? Would it throw an error on invalid order or just return empty response?

File: packages/drops/src/types/input.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
...
 export interface FetchDropsInput {
   limit: number;
   offset: number;
-  order?: string;
+  order?: Order;
   name?: string;
-  nameOrder?: string;
-  idOrder?: string;
+  nameOrder?: Order;
+  idOrder?: Order;
```
Comment (jm42) [inline packages/drops/src/types/input.ts:9]: What are these used for? What is the difference between `order` and `nameOrder` or `idOrder`?

File: packages/poaps/README.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# @poap-xyz/poaps
+
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](https://opensource.org/licenses/MIT)
+
+@poap-xyz/poaps is a package to interact with POAPs Tokens.
```
Comment (jm42) [inline packages/poaps/README.md:5]:
```suggestion
`@poap-xyz/poaps` is a package to fetch information about POAPs tokens, their on-chain data, their related drop and metadata.
```

File: packages/poaps/README.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+## Features
+
+- Fetch a single POAP
+- Fetch multiple POAPs
```
Comment (jm42) [inline packages/poaps/README.md:10]:
```suggestion
- Fetch a single POAP token,
- Search over minted POAPs by their collector or the drop they belong.
```

File: packages/poaps/src/PoapsClient.ts [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+      variables,
+    );
+
+    const poaps = data.poaps.map((poap) => {
+      const { drop } = poap;
+      return new POAP({
+        ...poap,
+        ...drop,
+        id: Number(poap.id),
+        minted_on: new Date(poap.minted_on),
```
Comment (jm42) [inline packages/poaps/src/PoapsClient.ts:77]: What the difference between `minted_on` and `created_date`?

Isn't `expiry_date` missing here?

File: packages/poaps/src/queries/PaginatedPoaps.ts [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+      id
+      minted_on
+      transfer_count
+      drop {
+        image_url
+        city
+        country
+        description
+        start_date
+        end_date
```
Comment (jm42) [inline packages/poaps/src/queries/PaginatedPoaps.ts:21]: What about `expiry_date`?

File: packages/poaps/src/queries/PaginatedPoaps.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    $offset: Int!
+    $orderBy: [poaps_order_by!]
+    $where: poaps_bool_exp
+  ) {
+    poaps(limit: $limit, offset: $offset, order_by: $orderBy, where: $where) {
+      chain
+      collector_address
+      drop_id
+      id
+      minted_on
```
Comment (jm42) [inline packages/poaps/src/queries/PaginatedPoaps.ts:13]: Is this called `created_date` or `minted_on`? I am missing something I think.

File: packages/poaps/src/queries/PaginatedPoaps.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  transfer_count: number;
+  minted_on: number;
+  drop_id: number;
+  drop: {
+    image_url: string;
+    city: string;
+    country: string;
+    description: string;
+    start_date: string;
+    end_date: string;
```
Comment (jm42) [inline packages/poaps/src/queries/PaginatedPoaps.ts:40]: `expiry_date` missing.

File: packages/poaps/src/queries/utils.ts [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export function createBetweenFilter(
+  key: string,
+  from?: string,
+  to?: string,
+): Record<string, any> {
+  const dateFilter: { _gte?: string; _lte?: string } = {};
+  if (from) dateFilter._gte = from;
+  if (to) dateFilter._lte = to;
```
Comment (jm42) [inline packages/poaps/src/queries/utils.ts:39]:
```suggestion
  if (from) {
    dateFilter._gte = from;
  }
  if (to) {
    dateFilter._lte = to;
  }
```

File: packages/poaps/tsconfig.json [hunk 15]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./dist",
+    "rootDir": "./src",
+    "composite": true
+  },
+  "include": ["./src/**/*.ts"]
+}
```
Comment (jm42) [inline packages/poaps/tsconfig.json:9]:
```suggestion
}

```

File: packages/poaps/src/PoapsClient.ts (hunk 10 above)
Comment (rlajous) [reply packages/poaps/src/PoapsClient.ts:77]: The expiry date is more information about the drop not the POAP itself i think

File: packages/poaps/src/queries/PaginatedPoaps.ts (hunk 12 above)
Comment (rlajous) [reply packages/poaps/src/queries/PaginatedPoaps.ts:13]: It's `minted_on`

File: packages/poaps/src/queries/PaginatedPoaps.ts (hunk 13 above)
Comment (rlajous) [reply packages/poaps/src/queries/PaginatedPoaps.ts:40]: The `expiry_date` is more information about the drop, not the POAP itself I think.

File: packages/poaps/src/PoapsClient.ts (hunk 10 above)
Comment (jm42) [reply packages/poaps/src/PoapsClient.ts:77]: Right.

File: packages/poaps/src/queries/PaginatedPoaps.ts (hunk 11 above)
Comment (jm42) [reply packages/poaps/src/queries/PaginatedPoaps.ts:21]: Alright.

File: packages/poaps/src/queries/PaginatedPoaps.ts (hunk 12 above)
Comment (jm42) [reply packages/poaps/src/queries/PaginatedPoaps.ts:13]: :+1:

File: packages/poaps/src/queries/PaginatedPoaps.ts (hunk 13 above)
Comment (jm42) [reply packages/poaps/src/queries/PaginatedPoaps.ts:40]: :+1:

File: .gitignore [hunk 16]
```diff
@@ -11,4 +11,5 @@ docs/.yarn/
 !.yarn/versions
 .next
 .idea
-.vscode
\ No newline at end of file
+.vscode
+*.txt
```
Comment (jm42) [inline .gitignore:15]: This is not the way, better to ignore the file that is know the test to output to.

File: .gitignore [hunk 17]
```diff
@@ -11,4 +11,5 @@ docs/.yarn/
 !.yarn/versions
 .next
 .idea
-.vscode
\ No newline at end of file
+.vscode
```
Comment (jm42) [inline .gitignore]: Kind of against adding your OS and IDE files to the ignore of everyone using the project with other things.

File: packages/drops/src/queries/utils.ts [hunk 18]
```diff
@@ -11,6 +11,13 @@ export function filterUndefinedProperties<T extends Record<string, any>>(
   return filteredObj;
 }
 
+export function creatUndefinedOrder(
```
Comment (jm42) [inline packages/drops/src/queries/utils.ts:14]: Time to start thinking about putting this utils somewhere sharable.

**Review by alavarello — APPROVED** [review]

**Pull request merged**